### PR TITLE
PicketLink Quickstarts

### PR DESCRIPTION
**Pull request by pedroigor**
This PR provides some PicketLink quickstarts. For a complete list of them and also the corresponding JIRAs, please access:

https://issues.jboss.org/issues/?jql=project%20%3D%20JDF%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20in%20(Quickstarts%2C%20quickstarts)%20AND%20labels%20%3D%20PicketLink%20ORDER%20BY%20priority%20DESC

Please, do not merge this PR until the PicketLink version is updated to the next version in JBoss BOMs. Once it is updated, I'll change the version of some dependencies and let you know when it is ready to be merged.

### CODE REVIEW

File: picketlink-authentication-http-basic/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+Target Product: EAP
+Source: <https://github.com/jboss-jdf/jboss-as-quickstart/>
+
+
+What is it?
+-----------
+
+This example demonstrates the use of *CDI 1.0* and *PicketLink* in *JBoss Enterprise Application Platform 6* or *JBoss AS 7*.
+
+You'll learn from this quickstart how to use PicletLink to authenticate users using the HTTP Basic scheme.
```
Comment (pmuir) [inline picketlink-authentication-http-basic/README.md]: PicletLink?

File: picketlink-authentication-http-basic/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+The application is configured to provide public access for some resources(eg.: /index.html) and to protected others for
+authenticated users only(eg.: /protected/*).
+
+Identity data such as users, roles and groups, are managed using PicketLink IDM backed by a file-based identity store.
+This store is used by default when no specific configuration is provided.
+
+*Note that HTTP Basic authentication authentication provides a low level of security because credentials are
+not encrypted before they are sent across a network. You should consider using it in conjunction with SSL or use
```
Comment (pmuir) [inline picketlink-authentication-http-basic/README.md:25]: Should we show in the QS how to use SSL?

**Comment by pmuir** [comment]
This basically looks great. I do want to see the two factor auth in action though!

I also wonder if we want to base all the quickstarts on jsf? Could be good to base some eg. on REST+JS?

**Comment by pedroigor** [comment]
I think we have some quickstarts in our backlog that requires REST+JS. 

Or do you to want to change some of the examples provided by this PR to use REST+JS ?

File: picketlink-authentication-http-basic/README.md (hunk 2 above)
Comment (pedroigor) [inline picketlink-authentication-http-basic/README.md:25]: That's a good idea. Maybe a link for the SSL configuration in the EAP documentation ?

File: picketlink-authentication-http-basic/README.md (hunk 1 above)
Comment (pedroigor) [inline picketlink-authentication-http-basic/README.md]: Fixed for all quickstarts.

**Comment by sgilda** [comment]
I am able to compile, deploy, and run the `picketlink-authentication-http-basic` quickstart successfully. 

When I attempt to compile the `picketlink-authentication-http-digest` quickstart, I get the following errors:

[ERROR]   The project org.jboss.as.quickstarts:jboss-as-picketlink-authentication-http-digest:7.1.2-SNAPSHOT (/home/sgilda/GitRepos/quickstart-jdf/picketlink-authentication-http-digest/pom.xml) has 7 errors
[ERROR]     Non-resolvable import POM: Could not find artifact org.jboss.bom:jboss-javaee-6.0-with-security:pom:1.0.7-SNAPSHOT @ line 71, column 19 -> [Help 2]
[ERROR]     'dependencies.dependency.version' for org.jboss.spec.javax.ejb:jboss-ejb-api_3.1_spec:jar is missing. @ line 85, column 17
[ERROR]     'dependencies.dependency.version' for javax.enterprise:cdi-api:jar is missing. @ line 93, column 17
[ERROR]     'dependencies.dependency.version' for org.jboss.spec.javax.annotation:jboss-annotations-api_1.1_spec:jar is missing. @ line 101, column 17
[ERROR]     'dependencies.dependency.version' for org.jboss.spec.javax.faces:jboss-jsf-api_2.1_spec:jar is missing. @ line 109, column 17
[ERROR]     'dependencies.dependency.version' for org.picketlink:picketlink-api:jar is missing. @ line 117, column 17
[ERROR]     'dependencies.dependency.version' for org.picketlink:picketlink-impl:jar is missing. @ line 124, column 17

I see a similar error when I try to compile the `picketlink-authentication-two-factor` quickstart.

I don't see any type of special set up required in the README. Am I forgetting to do something?

**Comment by pedroigor** [comment]
Hi sgilda,

```
Some of the quickstarts require the latest changes from PicketLink, which are target to beta4. As you can see, the error is about a 1.7.0-SNAPSHOT version of the JBoss Security BOM, which I have manually installed in my local repo changing the PicketLink version.

I'm not sure how we can get you running those quickstarts, without releasing PicketLink first and changing the BOMs properly.

Another solution for this is to follow this steps:

    1) Clone git://github.com/picketlink/picketlink.git

    2) Clone git@github.com:jboss-jdf/jboss-bom.git

    3) Build PicketLink (mvn clean install)

    4) Change JBoss BOMs to use PicketLink 2.5.0-SNAPSHOT

    5) Build JBoss BOMs

I know this is not cool, just telling you what I did to get the quickstarts running with the latest changes from PicketLink.
```

Thanks!
Pedro Igor

**Comment by sgilda** [comment]
@pedroigor : I don't think we can expect our users to do all this. 

@pmuir : I don't think we can merge until this is resolved. What do you think?

**Comment by pedroigor** [comment]
Sure, agreed. 

Indeed, this was my first comment to this PR :). As @pmuir asked.

**Comment by sgilda** [comment]
@pedroigor : Ping me when the BOMs are updated and I can test this again.

**Comment by pedroigor** [comment]
@sgilda : That's the idea, I'll let you know. Thanks.

**Comment by pmuir** [comment]
@pedroigor you could publish the BOM snapshot to the jboss snapshot repo, and include in the POM if you want people to be able to run it...

**Comment by pedroigor** [comment]
@pmuir : I have talked with Anil and we are planning a PL timed release on monday. After the release, I can publish the BOM snapshot.

**Comment by pedroigor** [comment]
This PR is ready to be merged once JBoss BOM 1.0.7.CR4 is released with JDF-359.

**Comment by pmuir** [comment]
Merge

**Pull request closed without merging**